src/host/ConsoleArguments.cpp

```diff
@@ -515,9 +514,10 @@ void ConsoleArguments::s_ConsumeArg(_Inout_ std::vector<std::wstring>& args, _In
             s_ConsumeArg(args, i);
             hr = S_OK;
         }
-        else if (arg == WIN32_INPUT_MODE)
+        else if (std::regex_match(arg, std::wregex(L"--[A-Za-z0-9]+")))
         {
-            _win32InputMode = true;
+            // Any other alphanumeric sequence starting with -- is likely
+            // an unrecognized argument that we need to ignore.
             s_ConsumeArg(args, i);
             hr = S_OK;
         }

```

Comment:
I'm not sure if you'll be happy with using a regex here, if that's going to add bloat to conhost. It looks to me like we're already using regex searches in the `TextBuffer`, but I'm not sure if that actually gets linked into conhost.

Worst case, we could make this only ignore the defunct `--win32input` option, but I thought it would be nice to have it dropping all the unsupported args, since there was already a TODO for that a few lines down from here.

Reply:
It's not linked into conhost as far as I know. I personally would like to replace it with ICU in the future so that we can get proper Unicode aware regex queries. ICU is also _significantly_ faster, like in the order >100x. It can go through the entire 120x9001 text buffer full of enwik8 in <3ms with a worst-case query like `e` (or some other common ASCII character).

Is there any reason to not just check if the `arg` starts with `--` and then ignore it? What other arguments could start with `--` and not be followed by a-z0-9?

Reply:
This, I am somewhat worried about. We go from explicitly rejecting unknowns to explicitly accepting them...

Since conpty.dll and openconsole.exe ship in lock-step, I would be comfortable just yanking all support for the argument and the special flag and everything. :)

After all . . . it never made it up to our public API surface on Windows!

Reply:
This got me thinking, and I'd be interested in what you and Leonard and @zadjii-msft think.

_This is speculative and not for this PR._

"In the future", we might want to add more types of conpty flags. Maybe. I don't love it, but maybe we will.

Mike and Michael used to prefer `--real --arguments` that set flags, rather than `--flags 0xABCD` to bulk enable all of `0xABCD`. The argument was that it was easier spot in a debugger or a process list.

I still think there's value in having `--flags 0xABCD`, but this PR made me realize that maybe what we actually want is `--flags.required 0x0001 --flags.optional 0x0002`. Conforming versions of conhost can say, "I don't support flag 0x0001" if `0x0001` is too new for their blood... but they can ignore `0x0002` if they don't know what it is.

Is that YAGNI?

Reply:
> Is there any reason to not just check if the `arg` starts with `--` and then ignore it? What other arguments could start with `--` and not be followed by a-z0-9?

That was actually my first attempt, but it caused this test case to fail:
https://github.com/microsoft/terminal/blob/c9e993a38ec04833e15edac230300948a0fab83b/src/host/ut_host/ConsoleArgumentsTests.cpp#L113

So then I thought I should maybe also check that the arg contained no spaces, but that still caused this test to fail:
https://github.com/microsoft/terminal/blob/c9e993a38ec04833e15edac230300948a0fab83b/src/host/ut_host/ConsoleArgumentsTests.cpp#L182

And that's when I figured I'd need to go with a regex.

But maybe those test cases aren't really something we need to support. I don't know.

Reply:
> Since conpty.dll and openconsole.exe ship in lock-step, I would be comfortable just yanking all support for the argument and the special flag and everything. :)

Won't this potentially break third-party terminals though? If this update eventually ships as the inbox conhost, and there are third-party terminals launching it with the `--win32input` parameter, then that's surely going to kill them.

I was also concerned by the wording of the comments regarding defterm here:
https://github.com/microsoft/terminal/blob/c9e993a38ec04833e15edac230300948a0fab83b/src/host/srvinit.cpp#L606-L609

That made me think with defterm you might have an old version of conhost being started that would hand off to a newer version of Windows Terminal. Or does it not work like that?

Reply:
Ah, that! That one doesn't actually matter, and is an example of one of my least favorite types of code[^1]. That code runs in the _receiving_ conhost (the one that's built out of this repo) and it's actually just using that string to tell another part of the same process how to act.

We can safely remove it, so long as win32 input mode really does default to always-requested.

[^1]: using strings and serialization to communicate state _in the same address space, between two components that could exchange objects or POD types_

Reply:
OK, so the plan then is to remove all the `--win32input` parameters when starting the conpty connection, and drop the regex check for invalid args - we can figure out how best to handle that at some point in the future.

But I need to ask again, are we sure this won't break any third-party terminals?

Reply:
> But I need to ask again, are we sure this won't break any third-party terminals?

Fffffair, and sorry I didn't answer it at first. My official stance is, _"we haven't built the conpty API for use without conpty.dll"_; if somebody is invoking it directly and passing their own pipes, they are allowed to keep the pieces when it breaks.

The command line args are effectively an internal interface. Technically. I know somebody's caught us out for breaking them before, but that's because they were setting "`conhost --pty`" as their _debugger_ to stop certain Windows processes from starting . . .

Reply:
OK, that's just me not understanding how it works - that seems reasonable. But do I keep the `PSEUDOCONSOLE_WIN32_INPUT_MODE` flag, and just not do anything with it, or should that be nuked as well?

Reply:
Ah, it would be fine for you to remove that flag in its entirety. :)

Reply:
That's the other half of the equation. We haven't made a publicly supported API-stable release of ConPTY's new undocked interface, so anybody using it is taking on `0.x` version level support. I'm happy to be the one with whom the buck stops if they come to ask!

Reply:
I think this is done now. I've dropped the regex test for the invalid args, and removed all the `--win32input` command line parameters as well as the `PSEUDOCONSOLE` flag. Everything still appears to work.